typing/typetexp.ml

```diff
@@ -275,6 +276,7 @@ and transl_type_aux env policy styp =
             | Longident.Lapply(_, _) -> fatal_error "Typetexp.transl_type"
           in
           let path, decl = Env.find_type_by_name lid2 env in
+          ignore(Env.lookup_cltype ~loc:lid.loc lid.txt env);
           (path, decl, false)
         with Not_found ->
           ignore (Env.lookup_class ~loc:lid.loc lid.txt env); assert false
```

Comment:
@lpw25, shouldn't this be using `Env.lookup_cltype ~loc:lid.loc lid.txt env)` as well ?

Reply:
I think you are right. Which shows that the manual:

  https://caml.inria.fr/pub/docs/manual-ocaml/types.html

is currently wrong. It says `class-path` and talks about classes rather than class types.

Reply:
Note that the manual is half-right : #t works for a class type `t`, but it only accepts class paths (e.g. `#A.c`)  and not all class type paths like `#F(X).t`.

Reply:
It looks like the code in the type-checker (modulo the change Hugo suggested) can handle any class path but the parser doesn't allow extended paths. We should probably change the parser from `class_longident` to `clty_longident` and update the documentation accordingly.

Reply:
I agree, I am planning to send a PR with the parser and documentation change next week.